stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/Call.kt

```diff
@@ -90,6 +90,9 @@ public fun <T : Any> Call<T>.doOnStart(scope: CoroutineScope, function: suspend
 public fun <T : Any> Call<T>.doOnResult(scope: CoroutineScope, function: suspend (Result<T>) -> Unit): Call<T> =
     DoOnResultCall(this, scope, function)
 
+public fun <T: Any> Call<T>.withPrecondition(scope: CoroutineScope, precondition: suspend () -> Result<Unit>): Call<T> =
```

Comment:
Should we mark it at Experimental and InternalStreamApi for now?

Reply:
I marked it with InternalStreamChatApi